app/views/RoomInfoEditView/index.tsx

```diff
@@ -339,7 +337,7 @@ class RoomInfoEditView extends React.Component<IRoomInfoEditViewProps, IRoomInfo
 			log(e);
 		}
 
-		await this.setState({ saving: false });
+		EventEmitter.emit(LOADING_EVENT, { visible: false });
```

Comment:
Why, instead of repeating this logic of importing so many separate things every time, not create a generic handler into loading container? Something like that?

`export const sendLoadingEvent = (value: boolean): void => EventEmitter.emit(LOADING_EVENT, { visible: value });`

In the long run it would be less expensive, since it is two less imports and would keep the logic in a single function and in a single file.